### PR TITLE
DOCS-595: Add C++ SDK

### PR DESCRIPTION
**Pull request by npentrel**

### CODE REVIEW

File: docs/program/sdks/_index.md [hunk 1]
```diff
@@ -20,11 +21,12 @@ Viam offers software development kits (SDKs) in popular languages which
 
 Use the SDK of your preferred language to write code to control your robots.
 
-Viam currently offers SDKs for the following three languages:
+Viam currently offers SDKs for the following languages:
 
 - [Python SDK](https://python.viam.dev/)
 - [Go SDK](https://pkg.go.dev/go.viam.com/rdk)
 - [TypeScript SDK](https://ts.viam.dev/)
+- [C++ SDK](https://cpp.viam.dev/)
```
Comment (andf-viam) [inline docs/program/sdks/_index.md:29]: opt: should we note "alpha" here? Perhaps cumbersome to remember to change later, but without such notice, these SDKs seem like equal citizens in this list.

File: docs/program/sdks/_index.md [hunk 2]
```diff
@@ -204,6 +205,43 @@ async function main() {
 main();
 ```
 
+{{% /tab %}}
+{{% tab name="C++" %}}
```
Comment (andf-viam) [inline docs/program/sdks/_index.md:209]: Noticed: selecting one tab on page does not auto-switch to same-name tabs everywhere on page. I can look into this!

File: docs/program/sdks/_index.md [hunk 3]
```diff
@@ -215,19 +253,27 @@ Find documentation on how to use these methods here:
...
 - [Sensor](/components/sensor/#api)
 - [Servo](/components/servo/#api)
-<!--
-Board API Docs should be added soon on Viam Documentation, + Encoder
--->
 
-You can find example code in the [Python SDK example GitHub repository](https://github.com/viamrobotics/viam-python-sdk/tree/main/examples/server/v1), the [Golang SDK example GitHub repository](https://github.com/viamrobotics/rdk/tree/main/examples), or the [TypeScript SDK example GitHub repository](https://github.com/viamrobotics/viam-typescript-sdk/tree/main/examples).
+{{% alert title="Tip" color="tip" %}}
+
+You can find example code using the SDKs in the <file>examples</file> folder of the respective repository:
```
Comment (andf-viam) [inline docs/program/sdks/_index.md:269]: nit: the `python` example takes us to a few subdirectories within the `examples` directory, while all other links take us directly to the `examples` directory. None of the four really require the additional step "then navigate to the `examples` folder" that this list intro seems to suggest. Perhaps just plainly stating without monospace helps to be precise for all four, something like:

> Each SDK provides example code in the following locations:

Or similar.

File: docs/program/sdks/_index.md [hunk 4]
```diff
@@ -215,19 +253,27 @@ Find documentation on how to use these methods here:
...
+
+- [Python SDK repository](https://github.com/viamrobotics/viam-python-sdk/tree/main/examples/server/v1)
+- [Go SDK repository](https://github.com/viamrobotics/rdk/tree/main/examples)
+- [TypeScript SDK repository](https://github.com/viamrobotics/viam-typescript-sdk/tree/main/examples)
+- [Viam C++ SDK repository](https://github.com/viamrobotics/viam-cpp-sdk/tree/main/src/viam/examples)
```
Comment (andf-viam) [inline docs/program/sdks/_index.md:274]: If you agree with above, recommend changing link titles to "X SDK examples" instead.

File: docs/program/sdks/_index.md [hunk 5]
```diff
@@ -254,6 +300,11 @@ go run ~/myCode/myViamFile.py
 
 For an example, see [this execution demo.](https://github.com/viamrobotics/viam-typescript-sdk/tree/main/examples/vanilla)
 
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+For an example, see [this execution demo.](https://github.com/viamrobotics/viam-typescript-sdk/tree/main/examples/vanilla)
```
Comment (andf-viam) [inline docs/program/sdks/_index.md:306]: This is the TS example. I wasn't able to guess a likely C++ example from the upstream repo; not sure what this should link to.

File: docs/program/sdks/_index.md [hunk 6]
```diff
@@ -204,6 +205,43 @@ async function main() {
...
+# include <viam/api/common/v1/common.pb.h>
+# include <viam/api/robot/v1/robot.grpc.pb.h>
+
+# include <viam/sdk/robot/client.hpp>
+# include <viam/sdk/components/camera/client.hpp>
+
+using namespace viam::sdk;
+
+int main() {
+  std::string host("guardian-main.vw3iu72d8n.viam.cloud");
```
Comment (stuqdog) [inline docs/program/sdks/_index.md:226]: I don't think we should include an actual uri/credentials here, should we?

File: docs/program/sdks/_index.md (hunk 5 above)
Comment (stuqdog) [inline docs/program/sdks/_index.md:306]: This looks like a typescript link. Probably the best link for C++ would be https://github.com/viamrobotics/viam-cpp-sdk/tree/main/src/viam/examples/dial, though we don't have a great README for it at this time.

**Review by stuqdog — CHANGES_REQUESTED** [review]
I don't see a link to C++ documentation from the `SDK DOCUMENTATION` pulldown menu at the top of the docs page, can we add that as well?

Otherwise a couple minor comments but this looks great, thanks so much for adding it!

File: docs/program/sdks/_index.md [hunk 7]
```diff
@@ -204,6 +205,43 @@ async function main() {
 main();
 ```
 
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+```cpp {class="line-numbers linkable-line-numbers"}
```
Comment (andf-viam) [inline docs/program/sdks/_index.md:211]: nit: `cpp` doesn't appear to highlight, and stands out as missing against the other three. Seems to be [the right one](https://gohugo.io/content-management/syntax-highlighting/) though (or maybe `c++` ?). I can investigate.

Comment (andf-viam) [reply docs/program/sdks/_index.md:211]: I created https://github.com/viamrobotics/docs/pull/1268 which I believe adds support for C++ highlighting. Let me know if I missed something!

File: docs/program/sdks/_index.md (hunk 1 above)
Comment (npentrel) [reply docs/program/sdks/_index.md:29]:
```suggestion
- [C++ SDK (alpha)](https://cpp.viam.dev/)
```

File: docs/program/sdks/_index.md (hunk 2 above)
Comment (npentrel) [inline docs/program/sdks/_index.md:209]:
```suggestion
{{% tab name="C++" %}}

{{< alert title="Note" color="note" >}}
The C++ SDK is currently in alpha.
{{< /alert >}}
```

File: docs/program/sdks/_index.md (hunk 7 above)
Comment (npentrel) [reply docs/program/sdks/_index.md:211]: thank you!

File: docs/program/sdks/_index.md [hunk 8]
```diff
@@ -204,6 +205,43 @@ async function main() {
...
+
+# include <viam/sdk/robot/client.hpp>
+# include <viam/sdk/components/camera/client.hpp>
+
+using namespace viam::sdk;
+
+int main() {
+  std::string host("guardian-main.vw3iu72d8n.viam.cloud");
+  DialOptions dial_opts;
+  Credentials credentials("gf1aufai3dr0b5o9csko93b1d8llfwolisa0my141wlmvkmg");
```
Comment (npentrel) [inline docs/program/sdks/_index.md:228]:
```suggestion
  Credentials credentials("SECRET FROM THE VIAM APP");
```

File: docs/program/sdks/_index.md (hunk 6 above)
Comment (npentrel) [reply docs/program/sdks/_index.md:226]:
```suggestion
  std::string host("ADDRESS FROM THE VIAM APP");
```

File: docs/program/sdks/_index.md (hunk 3 above)
Comment (npentrel) [reply docs/program/sdks/_index.md:273]:
```suggestion
Each SDK provides example code in its repository:
```

File: docs/program/sdks/_index.md [hunk 9]
```diff
@@ -215,19 +257,27 @@ Find documentation on how to use these methods here:
...
 
-You can find example code in the [Python SDK example GitHub repository](https://github.com/viamrobotics/viam-python-sdk/tree/main/examples/server/v1), the [Golang SDK example GitHub repository](https://github.com/viamrobotics/rdk/tree/main/examples), or the [TypeScript SDK example GitHub repository](https://github.com/viamrobotics/viam-typescript-sdk/tree/main/examples).
+{{% alert title="Tip" color="tip" %}}
+
+You can find example code using the SDKs in the <file>examples</file> folder of the respective repository:
+
+- [Python SDK repository](https://github.com/viamrobotics/viam-python-sdk/tree/main/examples/server/v1)
+- [Go SDK repository](https://github.com/viamrobotics/rdk/tree/main/examples)
+- [TypeScript SDK repository](https://github.com/viamrobotics/viam-typescript-sdk/tree/main/examples)
+- [Viam C++ SDK repository](https://github.com/viamrobotics/viam-cpp-sdk/tree/main/src/viam/examples)
```
Comment (npentrel) [inline docs/program/sdks/_index.md:278]:
```suggestion
- [Python SDK examples](https://github.com/viamrobotics/viam-python-sdk/tree/main/examples/server/v1)
- [Go SDK examples](https://github.com/viamrobotics/rdk/tree/main/examples)
- [TypeScript SDK examples](https://github.com/viamrobotics/viam-typescript-sdk/tree/main/examples)
- [Viam C++ SDK examples](https://github.com/viamrobotics/viam-cpp-sdk/tree/main/src/viam/examples)
```

File: docs/program/sdks/_index.md [hunk 10]
```diff
@@ -254,6 +304,11 @@ go run ~/myCode/myViamFile.py
 
 For an example, see [this execution demo.](https://github.com/viamrobotics/viam-typescript-sdk/tree/main/examples/vanilla)
 
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+For an example, see [this execution demo.](https://github.com/viamrobotics/viam-typescript-sdk/tree/main/examples/vanilla)
```
Comment (npentrel) [inline docs/program/sdks/_index.md:310]:
```suggestion
For information on running C++ code see [here](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/BUILDING.md).
```

File: docs/program/sdks/_index.md (hunk 5 above)
Comment (npentrel) [reply docs/program/sdks/_index.md:310]: I think it actually should be https://github.com/viamrobotics/viam-cpp-sdk/blob/main/BUILDING.md again. We're looking for something that points to runnign the code

**Review by andf-viam — APPROVED** [review]
LGTM! Nice work!

**Review by stuqdog — APPROVED** [review]
LGTM!

**Pull request merged**